### PR TITLE
Feature/DEV-22-user-languages-page

### PR DESCRIPTION
**Pull request by RC-3222**
Added user languages tab/page (functionally simillar to the "skills" page)
image

image

image

### CODE REVIEW

File: src/modules/users/components/languages-dialog/languages-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }));
+
+    const languageOptions = languages
+        .map((item) => ({
+            label: item.name,
+            value: item.name,
+        }))
+        .filter((lang) => !existingLanguages.includes(lang.label));
+
+    const [deleteMutation, { loading: deleteLoading }] = useDeleteUserLanguage();
```
Comment (shev7) [inline src/modules/users/components/languages-dialog/languages-dialog.tsx]: put all api hooks to the top

File: src/modules/users/components/languages-dialog/languages-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            variables: {
+                language: {
+                    userId: userId,
+                    name: selectedLanguage?.name ? [selectedLanguage.name] : [],
+                },
+            },
+        }).then(() => {
+            onClose();
+        });
+    };
```
Comment (shev7) [inline src/modules/users/components/languages-dialog/languages-dialog.tsx]: use onCompleted callback as  prop of deleteMutation

File: src/modules/users/components/languages-list/item/item.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Button, Flex } from "antd";
+import { LanguageProficiency } from "cv-graphql";
+import classNames from "classnames";
+
+import styles from "./item.module.scss";
+
+interface LanguagesListItem {
```
Comment (shev7) [inline src/modules/users/components/languages-list/item/item.tsx]:
```suggestion
interface LanguagesListItemProps {
```

File: src/modules/users/components/languages-list/languages-list.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import { LanguagesFormSchemaType } from "../schemas/languages";
+
+import styles from "./languages-list.module.scss";
+
+interface SkillProps {
+    userId: string;
+    canEdit: boolean;
+}
+
+const LanguagesList = ({ userId, canEdit }: SkillProps) => {
```
Comment (shev7) [inline src/modules/users/components/languages-list/languages-list.tsx]: Component
ComponentProps

follow this prop naming convention

File: src/modules/users/components/languages-list/languages-list.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                selectedLanguage,
+                existingLanguages: existingLanguageLabels,
+                initialValues: {
+                    name: selectedLanguage.name,
+                    proficiency: selectedLanguage.proficiency,
+                },
+                isSubmitting: loadingUpdate,
+            }),
+        [openLanguageDialog, userId, loadingUpdate, existingLanguageLabels]
+    );
```
Comment (shev7) [inline src/modules/users/components/languages-list/languages-list.tsx]: memoization is not even needed here cause you passing this callback to simple button component

File: src/modules/users/components/languages-list/languages-list.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    if (loading) return <Spin size="large" />;
+
+    return (
+        <Flex vertical className={styles.list}>
+            {canEdit ? (
+                <Button size={"large"} type="text" onClick={addLanguage}>
+                    <PlusOutlined />
+                    {t("languages.addLanguage")}
+                </Button>
+            ) : null}
```
Comment (shev7) [inline src/modules/users/components/languages-list/languages-list.tsx]:
```suggestion
            {canEdit && (
                <Button size={"large"} type="text" onClick={addLanguage}>
                    <PlusOutlined />
                    {t("languages.addLanguage")}
                </Button>
            ) }
```

avoid using ternaries

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/modules/users/components/languages-dialog/languages-dialog.tsx (hunk 1 above)
Comment (RC-3222) [reply src/modules/users/components/languages-dialog/languages-dialog.tsx]: fixed

File: src/modules/users/components/languages-dialog/languages-dialog.tsx (hunk 2 above)
Comment (RC-3222) [reply src/modules/users/components/languages-dialog/languages-dialog.tsx]: fixed

File: src/modules/users/components/languages-list/item/item.tsx (hunk 3 above)
Comment (RC-3222) [reply src/modules/users/components/languages-list/item/item.tsx]: fixed

File: src/modules/users/components/languages-list/languages-list.tsx (hunk 4 above)
Comment (RC-3222) [reply src/modules/users/components/languages-list/languages-list.tsx]: fixed

File: src/modules/users/components/languages-list/languages-list.tsx (hunk 5 above)
Comment (RC-3222) [reply src/modules/users/components/languages-list/languages-list.tsx]: fixed

File: src/modules/users/components/languages-list/languages-list.tsx (hunk 6 above)
Comment (RC-3222) [reply src/modules/users/components/languages-list/languages-list.tsx]: fixed

**Review by shev7 — APPROVED** [review]

**Review by AAKlimkov — APPROVED** [review]

**Review by savelievamareena — APPROVED** [review]

**Pull request merged**